src/mixer.cc

```diff
@@ -123,11 +124,32 @@ void Mixer::Tick() {
       output.Push(m);
     }
   }
+  cyclus::toolkit::RecordTimeSeries<double>("supply"+out_commod, this, output.quantity());
 }
 
 std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
 Mixer::GetMatlRequests() {
   using cyclus::RequestPortfolio;
+  using cyclus::toolkit::RecordTimeSeries;
+
+  for (int i = 0; i < mixing_ratios.size(); i++)
+  {
+    std::string name = "in_stream_" + std::to_string(i);
+    std::map<std::string, double>::iterator it;
+    std::map<std::string, double>::iterator max_it = in_commods[i].begin();
+    double prev_pref = 0;
+    for (it = in_commods[i].begin(); it != in_commods[i].end(); it++)
+    {
+      double pref = it->second;
```

Comment:
How come you don't do the fancy std::max_element / std::distance thing here?

Reply:
I don't think that's applicable here.. is it?